.github/workflows/auto-fix.yml

```diff
@@ -0,0 +1,59 @@
+name: Auto-fix Code Issues
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  auto-fix:
+    runs-on: ubuntu-latest
+    # Skip if commit message contains Auto-fix:
+    if: |
+      (github.event_name == 'push' && !contains(github.event.head_commit.message, 'Auto-fix:')) ||
+      (github.event_name == 'pull_request' && !contains(github.event.pull_request.head.commit.message, 'Auto-fix:'))
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v6
+        with:
+          ref: ${{ github.head_ref || github.ref }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v21
+
+      - name: Setup Nix cache
+        uses: DeterminateSystems/magic-nix-cache-action@v13
+
+      - name: Run statix fix
+        run: |
+          nix run nixpkgs#statix -- fix .
+
+      - name: Run formatter
+        run: nix fmt .
+
+      - name: Check for uncommitted changes on fork PRs
```

Comment:
Turns out `nix flake check` in the other workflow will run all of the git-hook checks, so we can just block on that workflow and don't need this step. In fact, we can just skip this whole workflow if we're dealing with a fork PR.